### DISCUSSION TITLE
LEFT ALT - disable focusing on a menu bar

### DISCUSSION THREAD

**Discussion by Yahuma** [freeplane/freeplane#402] [Help]
Hello,

I'm using LEFT ALT in global hotkeys for a video player to pause and rewind video during making notes.
During work focus very often accidentally goes to the menu bar which slows down studying.
I would like to avoid changing shortcuts because these with LEFT ALT are very ergonomic.

Is it possible to disable LEFT ALT behavior which move focus to a menu bar?

Best Regards,
Bartek

**Comment by euu2021** [comment]
As a workaround, try using the map in an undocked window (`View->New map view` ; undock clicking on the little arrow in the tab). The LeftAlt won't bother you. Then, use the Command Search to get what you need from the menus.

The shortcoming is that you still can't use the LeftAlt for hotkeys. I mostly use undocked windows, but I would like to have LeftAlt+Spacebar as a hotkey, so I second your request.

**Reply by Yahuma** [reply]
Thank you for response.

You've inspired me and I've found even better solution!

It's possible to hide menu bar (right click on map and there is an option to disable it) then there is no problem with focus, but it doesn't solve your problem with LeftAlt+Space.

Mayby it is possible to make an option to enable/disable ALT behavior.

@eu2021 Thanks again for message.

**Reply by euu2021** [reply]
Hiding the menu bar has a shortcoming, though (I don't know how you will use, so I don't if it's relevant): if you are writing on the node core, pressing LeftAlt closes the core input area. On an undocked window, it doesn't happen.

**Comment by Yahuma** [comment]
@dpolivaev Thank you so much for this amazing program. It's still open on my computer. I have everything in it: my personal diary, notes, knowledge archive, ...

I don't want to take up much of your time, please answer as much as you know at the moment.

I've got problem like on the screen:

![Screenshot - 12 01 2023 , 07_03_00](https://user-images.githubusercontent.com/26204454/211990095-a1285959-28fb-4440-aa3b-7cb26648e730.png)

- Is there currently an option to disable this LEFT ALT key behavior, maybe some workaround?
- If not, is this technically possible to implement such a switch in options to disable this LEFT ALT key behavior. 
- If so, is there a lot of work to do with it?

Best Regards,
Bartek

**Reply by macmarrum** [reply]
Hi Bartek

You can hide the menu bar, thus preventing the `Alt` key bindings from being created: `View->Controls->Menubar`.

You can also bind a hot key to the action to toggle it, or use `Help->Command Search (Ctrl+F1)` > `menubar`.

**Comment by Yahuma** [comment]
Hi macmarrum,

Thank you for your response.

Hiding a menu bar doesn't solve the problem. When menu is hidden and I press ALT focus is still moved from node.

Binding a hot could be a workaround if it would be possible to assing action to the ALT itself, but it's not possible.

Best Regards,
Bartek

**Reply by macmarrum** [reply]
I just realized that it's different on Linux, which is the OS I am using, and on Windows, which is apparently what you are using.

I believe that on Windows `Alt` is interpreted by the operating system, e.g. `Alt-SPACE` opens the window context menu, `Alt` activates the menu bar. Therefore, there might be a way to disable it for the entire OS. The AutoHotkey solution looks promising: https://superuser.com/questions/309005/disable-the-activation-of-the-menu-bar-when-alt-is-pressed-in-windows-7

**Reply by dpolivaev** [reply]
I think macmarrum is right, check https://stackoverflow.com/a/1722944/1833472

**Reply by Yahuma** [reply]
@dpolivaev, @macmarrum,

Thank you so much. I've found solution with AutoHotkey.

This AHK 2.0 script solves problem with moving focus to the menu bar with ALT key, but it allows to using ALT key in other keyboard shortcuts:

```
LAlt & F24:: return
LAlt:: Send "{LAlt}"
```
